gcp/live/dev/spec/network_spec.rb

```diff
@@ -0,0 +1,168 @@
+require 'json'
+require 'rspec/expectations'
+
+class Pod
+  def self.from_hash(hash)
+    Pod.new(hash)
+  end
+
+  def initialize(hash)
+    @data = hash
+  end
+
+  def ip
+    @data['status']['podIP']
+  end
+
+  def name
+    @data['metadata']['name']
+  end
+
+  def has_label(key, value)
+    @data['metadata']['labels'].keys.include?(key) &&
+    @data['metadata']['labels'][key] == value
+  end
+end
+
+def get_gpii_pods
+  JSON.parse(`kubectl get pods -n gpii -o json`)['items']
+    .collect{ |item| Pod.from_hash(item) }
+end
+
+def get_gpii_pods_labeled(key, value)
+  get_gpii_pods
+    .select{ |item| item.has_label(key, value) }
+end
+
+def kubectl(cmd)
+  `kubectl #{cmd} 2>/dev/null`
+end
+
+describe "Security:" do
+  before :all do
+    @couchdb_pods = get_gpii_pods_labeled("app", "couchdb")
+    @preferences_pods = get_gpii_pods_labeled("app", "preferences")
+    @flowmanager_pods = get_gpii_pods_labeled("app", "flowmanager")
+    @all_pods = @couchdb_pods | @preferences_pods | @flowmanager_pods
+
+    @couchdb_pods.each do |pod|
+      kubectl("exec -n gpii -it #{pod.name} -- sh -c \"[ ! -f /bin/nc ] && apt-get update -y && apt-get install netcat -y\"")
+    end
+  end
+
+  context "Preferences Pod" do
+    context "Container Port 8081" do
+      it "listens on port 8081" do
+        @preferences_pods.each do |pod|
+          kubectl("exec -n gpii -it #{pod.name} -c preferences -- nc -z #{pod.ip} 8081")
```

Comment:
re. above - we can't expect nc to be present in containers

Reply:
How about downloading `busybox` and pushing it onto `/tmp/`? This will give us the ability to execute netcat without permanently altering that specific container.

Reply:
Same thing, you're polluting container & increasing its attack surface - doesn't really matter where you put it ;). And might not be possible with some containers as a general approach - currently, you're assuming all are alpine based, that you can write to a given destination. Also wouldn't work with containers that are not allowed to run stuff as root (another security feature we should use)